src/Compilers/CSharp/Test/Emit3/PartialEventsAndConstructorsTests.cs

```diff
@@ -1411,4 +1430,625 @@ void M()
+    }
+
+    [Fact]
+    public void Difference_Params()
+    {
+        var source = """
+            using System.Collections.Generic;
+            partial class C1
+            {
+                partial C1(object[] x);
+                partial C1(params object[] x) { }
+            }
+            partial class C2
+            {
+                partial C2(int x, params IEnumerable<object> y);
+                partial C2(int x, IEnumerable<object> y) { }
+            }
+            partial class C3
+            {
+                partial C3(params IEnumerable<object> x, int y);
+                partial C3(IEnumerable<object> x, int y) { }
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics(
+            // (5,13): error CS0758: Both partial member declarations must use a params parameter or neither may use a params parameter
+            //     partial C1(params object[] x) { }
+            Diagnostic(ErrorCode.ERR_PartialMemberParamsDifference, "C1").WithLocation(5, 13),
+            // (10,13): error CS0758: Both partial member declarations must use a params parameter or neither may use a params parameter
+            //     partial C2(int x, IEnumerable<object> y) { }
+            Diagnostic(ErrorCode.ERR_PartialMemberParamsDifference, "C2").WithLocation(10, 13),
+            // (14,16): error CS0231: A params parameter must be the last parameter in a parameter list
+            //     partial C3(params IEnumerable<object> x, int y);
+            Diagnostic(ErrorCode.ERR_ParamsLast, "params IEnumerable<object> x").WithLocation(14, 16));
+    }
+
+    [Fact]
+    public void Difference_ParameterNames()
+    {
+        var source = """
+            partial class C
+            {
+                partial C(int x);
+                partial C(int y) { }
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics(
+            // (4,13): warning CS9256: Partial member declarations 'C.C(int x)' and 'C.C(int y)' have signature differences.
+            //     partial C(int y) { }
+            Diagnostic(ErrorCode.WRN_PartialMemberSignatureDifference, "C").WithArguments("C.C(int x)", "C.C(int y)").WithLocation(4, 13));
+    }
+
+    [Fact]
+    public void Difference_OptionalParameters()
+    {
+        var source = """
+            partial class C1
+            {
+                partial C1(int x = 1);
+                partial C1(int x) { }
+            }
+            partial class C2
+            {
+                partial C2(int x);
+                partial C2(int x = 1) { }
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void Difference_DefaultParameterValues()
+    {
+        var source = """
+            partial class C
+            {
+                partial C(int x = 1);
+                partial C(int x = 2) { }
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics();
+    }
```

Comment:
>} [](http://example.com/codeflow?start=4&length=1)

Consider testing use, `new C()`, and reporting the value of the argument passed.